camel/societies/babyagi_playing.py

```diff
@@ -0,0 +1,291 @@
+                                        output_language)
+
+        sys_msg_generator = SystemMessageGenerator(
+            task_type=self.task_type, **(sys_msg_generator_kwargs or {}))
+
+        init_assistant_sys_msg = sys_msg_generator.from_dicts(
+            meta_dicts=[dict(task=self.task_prompt)],
+            role_tuples=[
+                (assistant_role_name, RoleType.ASSISTANT),
+            ],
+        )
+
+        self.assistant_agent: ChatAgent
+        self.assistant_sys_msg: BaseMessage
+        self.task_creation_agent: TaskCreationAgent
+        self.task_prioritize_agent: TaskPrioritizeAgent
+        self.init_agents(init_assistant_sys_msg[0], assistant_agent_kwargs,
+                         task_create_agent_kwargs,
+                         task_prioritize_agent_kwargs, output_language,
+                         message_window_size)
+
+        self.tasks: deque = deque([])
+        self.solved_tasks: List[str] = []
+        self.MAX_TASK_HISTORY = 10
+
+    def init_specified_task_prompt(
+            self, assistant_role_name: str, user_role_name: str,
+            task_specify_agent_kwargs: Optional[Dict],
+            extend_task_specify_meta_dict: Optional[Dict],
+            output_language: Optional[str]):
+        r"""Use a task specify agent to generate a specified task prompt.
+        Generated specified task prompt will be used to replace original
+        task prompt. If there is no task specify agent, specified task
+        prompt will not be generated.
+
+        Args:
+            assistant_role_name (str): The name of the role played by the
+                assistant.
+            user_role_name (str): The name of the role played by the user.
+            task_specify_agent_kwargs (Dict, optional): Additional arguments
+                to pass to the task specify agent.
+            extend_task_specify_meta_dict (Dict, optional): A dict to extend
+                the task specify meta dict with.
+            output_language (str, optional): The language to be output by the
+                agents.
+        """
+        if self.with_task_specify:
+            task_specify_meta_dict = dict()
+            if self.task_type in [TaskType.AI_SOCIETY, TaskType.MISALIGNMENT]:
+                task_specify_meta_dict.update(
+                    dict(assistant_role=assistant_role_name,
+                         user_role=user_role_name))
+            task_specify_meta_dict.update(extend_task_specify_meta_dict or {})
+            task_specify_agent = TaskSpecifyAgent(
+                task_type=self.task_type,
+                output_language=output_language,
+                **(task_specify_agent_kwargs or {}),
+            )
+            self.specified_task_prompt = task_specify_agent.run(
+                self.task_prompt,
+                meta_dict=task_specify_meta_dict,
+            )
+            self.task_prompt = self.specified_task_prompt
```

Comment:
Overwriting self.task_prompt does not seem right. Why do we need self.specified_task_prompt then?

Please avoid mutation of fields. Also initialize a field only in one method. Both self.task_prompt and self.specified_task_prompt must be initialized in one method: init_specified_task_prompt in this case.

Reply:
Thanks for your comments.

I noticed the self.specified_task_prompt would lead to more concrete task planning and better task execution result. Otherwise, the user-input task prompt is too high-level to get useful results. 

And, I avoid overwriting the self.task_prompt and initialize self.specified_task_prompt in init_specified_task_prompt method.